# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/RenderRootTypes.h
# Commit: b294575c279c
# Full Hash: b294575c279c9681d0a26b1592f47014262f1d03
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-07-12 09:56:21
# Regressor Bug: 1547351
# File Overlap Count: 2
# Description:
#   Bug 1547351 - Automatically set content process render root r=kats
#   
#   This implements the idea of automatically setting a content proc's
#   render root based on the render root enclosing the iframe that
#   points to it. There was a bit of cleanup in here that was a bit
# ==============================================================================

diff -r 69260373f95c -r b294575c279c gfx/layers/wr/RenderRootTypes.h
--- a/gfx/layers/wr/RenderRootTypes.h	Fri Jul 12 03:19:28 2019 +0000
+++ b/gfx/layers/wr/RenderRootTypes.h	Fri Jul 12 03:19:28 2019 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/webrender/WebRenderTypes.h"
 #include "mozilla/layers/WebRenderMessages.h"
 #include "mozilla/layers/WebRenderScrollData.h"
+#include "mozilla/Variant.h"
 
 namespace mozilla {
 
@@ -18,11 +19,13 @@
 
 struct RenderRootDisplayListData {
   wr::RenderRoot mRenderRoot;
+  wr::IdNamespace mIdNamespace;
   LayoutDeviceRect mRect;
   nsTArray<WebRenderParentCommand> mCommands;
   wr::LayoutSize mContentSize;
   Maybe<mozilla::ipc::ByteBuf> mDL;
   wr::BuiltDisplayListDescriptor mDLDesc;
+  nsTArray<wr::PipelineId> mRemotePipelineIds;
   nsTArray<OpUpdateResource> mResourceUpdates;
   nsTArray<RefCountedShmem> mSmallShmems;
   nsTArray<mozilla::ipc::Shmem> mLargeShmems;
@@ -36,8 +39,19 @@
   nsTArray<RefCountedShmem> mSmallShmems;
   nsTArray<mozilla::ipc::Shmem> mLargeShmems;
   ScrollUpdatesMap mScrollUpdates;
+  uint32_t mPaintSequenceNumber;
 };
 
+struct ResourceUpdates {
+  nsTArray<OpUpdateResource> mResourceUpdates;
+  nsTArray<RefCountedShmem> mSmallShmems;
+  nsTArray<mozilla::ipc::Shmem> mLargeShmems;
+};
+
+typedef Variant<RenderRootDisplayListData, RenderRootUpdates, ResourceUpdates,
+                FocusTarget>
+    RenderRootDeferredData;
+
 }  // namespace layers
 
 namespace ipc {